src/server/scripts/World/npcs_special.cpp

```diff
@@ -231,84 +242,95 @@ class npc_riggle_bassbait : public CreatureScript
     {
         npc_riggle_bassbaitAI(Creature* c) : ScriptedAI(c)
         {
-            events.Reset();
-            events.ScheduleEvent(EVENT_RIGGLE_ANNOUNCE, 1000, 1, 0);
-            finished = sWorld->getWorldState(GAME_EVENT_FISHING) == 1;
-            startWarning = false;
-            finishWarning = false;
+            m_uiTimer = 0;
+            auto prevWinTime = sWorld->getWorldState(STV_FISHING_PREV_WIN_TIME);
+            if (time(nullptr) - prevWinTime > DAY)
+            {
+                // reset all after 1 day
+                sWorld->setWorldState(STV_FISHING_ANNOUNCE_EVENT_BEGIN, 1);
+                sWorld->setWorldState(STV_FISHING_ANNOUNCE_POOLS_DESPAN, 0);
+                sWorld->setWorldState(STV_FISHING_HAS_WINNER, 0);
+            }
         }
 
-        EventMap events;
-        bool finished;
-        bool startWarning;
-        bool finishWarning;
-
-        uint32 GetData(uint32 type) const override
-        {
-            if (type == DATA_ANGLER_FINISHED)
-                return (uint32)finished;
-
-            return 0;
-        }
+        uint32 m_uiTimer;
 
-        void DoAction(int32 param) override
+        void CheckTournamentState() const
         {
-            if (param == DATA_ANGLER_FINISHED)
+            if (sGameEventMgr->IsActiveEvent(GAME_EVENT_FISHING_TURN_INS) && !sWorld->getWorldState(STV_FISHING_HAS_WINNER))
             {
-                finished = true;
-                sWorld->setWorldState(GAME_EVENT_FISHING, 1);
+                if (!me->IsQuestGiver())
+                {
+                    me->SetFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_QUESTGIVER);
+                }
+                if (sWorld->getWorldState(STV_FISHING_ANNOUNCE_EVENT_BEGIN))
+                {
+                    sCreatureTextMgr->SendChat(me, RIGGLE_SAY_START, 0, CHAT_MSG_MONSTER_YELL, LANG_UNIVERSAL, TEXT_RANGE_ZONE);
```

Comment:
Why not Say()?

Reply:
This should be a map wide announcement (at least from what I saw on VMangos).
I haven't checked whether the broadcast rage is currently correct (I left this code as it was before), maybe `TEXT_RANGE_ZONE` can be replaced with smth like `TEXT_RANGE_MAP`

Reply:
This can still be done with Say() as you determine the range in creature_text.TextRange

Reply:
https://www.azerothcore.org/wiki/creature_text#textrange

Reply:
Well I guess Yell() would be the proper function

Reply:
I had to use Talk() instead of Yell()/Say() since both functions did expect a string view instead of a text id.

Reply:
Nvm, yell works. Just when passing LANG_UNIVERSAL beside the text id there was no function found.